homeassistant/components/modern_forms/fan.py

```diff
@@ -0,0 +1,195 @@
+"""Support for Modern Forms Fan Fans."""
+from __future__ import annotations
+
+from functools import partial
+from typing import Any, Callable, cast
+
+from aiomodernforms.const import FAN_POWER_OFF, FAN_POWER_ON
+import voluptuous as vol
+
+from homeassistant.components.fan import SUPPORT_DIRECTION, SUPPORT_SET_SPEED, FanEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import callback
+import homeassistant.helpers.entity_platform as entity_platform
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.util.percentage import (
+    int_states_in_range,
+    percentage_to_ranged_value,
+    ranged_value_to_percentage,
+)
+
+from . import (
+    ModernFormsDataUpdateCoordinator,
+    ModernFormsDeviceEntity,
+    modernforms_exception_handler,
+)
+from .const import (
+    ATTR_SLEEP_TIME,
+    CLEAR_TIMER,
+    DOMAIN,
+    OPT_ON,
+    OPT_SPEED,
+    SERVICE_CLEAR_FAN_SLEEP_TIMER,
+    SERVICE_SET_FAN_SLEEP_TIMER,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities: Callable
+) -> None:
+    """Set up a Modern Forms platform from config entry."""
+
+    coordinator: ModernFormsDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        SERVICE_SET_FAN_SLEEP_TIMER,
+        {
+            vol.Required(ATTR_SLEEP_TIME): vol.All(
+                vol.Coerce(int), vol.Range(min=1, max=1440)
+            ),
+        },
+        "async_set_fan_sleep_timer",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_CLEAR_FAN_SLEEP_TIMER,
+        {},
+        "async_clear_fan_sleep_timer",
+    )
+
+    update_lights = partial(
```

Comment:
should this be `update_fan`, or maybe just `update_func`?

Reply:
sure